src/bun.js/api/server/StaticRoute.zig

```diff
@@ -0,0 +1,256 @@
+    this.onResponseComplete(resp);
+}
+
+fn onResponseComplete(this: *StaticRoute, resp: HTTPResponse) void {
+    resp.clearAborted();
+    resp.clearOnWritable();
+    resp.clearTimeout();
+
+    if (this.server) |server| {
+        server.onStaticRequestComplete();
+    }
+
+    this.deref();
+}
+
+pub fn doRenderBlob(this: *StaticRoute, resp: HTTPResponse, did_finish: *bool) void {
+    // We are not corked
+    // The body is small
+    // Faster to do the memcpy than to do the two network calls
+    // We are not streaming
+    // This is an important performance optimization
+    if (this.blob.fastSize() < 16384 - 1024) {
+        resp.corked(doRenderBlobCorked, .{ this, resp, did_finish });
+    } else {
+        this.doRenderBlobCorked(resp, did_finish);
+    }
+}
+
+pub fn doRenderBlobCorked(this: *StaticRoute, resp: HTTPResponse, did_finish: *bool) void {
+    this.renderMetadata(resp);
+    this.renderBytes(resp, did_finish);
+}
+
+fn onWritable(this: *StaticRoute, write_offset: u64, resp: HTTPResponse) void {
+    if (this.server) |server| {
+        resp.timeout(server.config().idleTimeout);
+    }
+
+    if (!this.onWritableBytes(write_offset, resp)) {
+        this.toAsync(resp);
+        return;
+    }
+
+    this.onResponseComplete(resp);
+}
+
+fn onWritableBytes(this: *StaticRoute, write_offset: u64, resp: HTTPResponse) bool {
+    const blob = this.blob;
+    const all_bytes = blob.slice();
+
+    const bytes = all_bytes[@min(all_bytes.len, @as(usize, @truncate(write_offset)))..];
+
+    if (!resp.tryEnd(
+        bytes,
+        all_bytes.len,
+        resp.shouldCloseConnection(),
+    )) {
+        return false;
+    }
+
+    return true;
+}
+
+fn doWriteStatus(_: *StaticRoute, status: u16, resp: HTTPResponse) void {
+    switch (resp) {
+        .SSL => |r| writeStatus(true, r, status),
+        .TCP => |r| writeStatus(false, r, status),
+    }
+}
+
+fn doWriteHeaders(this: *StaticRoute, resp: HTTPResponse) void {
+    switch (resp) {
+        inline .SSL, .TCP => |s| {
+            const entries = this.headers.entries.slice();
+            const names: []const Api.StringPointer = entries.items(.name);
+            const values: []const Api.StringPointer = entries.items(.value);
+            const buf = this.headers.buf.items;
+
+            for (names, values) |name, value| {
+                s.writeHeader(name.slice(buf), value.slice(buf));
+            }
+        },
+    }
+}
+
+fn renderBytes(this: *StaticRoute, resp: HTTPResponse, did_finish: *bool) void {
+    did_finish.* = this.onWritableBytes(0, resp);
+}
+
+fn renderMetadata(this: *StaticRoute, resp: HTTPResponse) void {
+    var status = this.status_code;
+    const size = this.cached_blob_size;
+
+    status = if (status == 200 and size == 0 and !this.blob.isDetached())
+        204
+    else
+        status;
+
+    this.doWriteStatus(status, resp);
+    this.doWriteHeaders(resp);
+}
+
+const StaticRoute = @This();
```

Comment:
nit: make this the first non-comment line. it makes it very obvious to a reader that the file is a struct for that name. not a big deal, one day i'll write a lint to force the ordering of import constants.